### PR TITLE
Allow multiple `root` routes in same scope level

### PR DESCRIPTION
**Pull request by rafaelsales**
TL;DR:
**Currently I have:**

``` ruby
get '/', to: 'portfolio#show', constraints: ->(req) { Hostname.portfolio_site?(req.host) }
get '/', to: 'blog#show',      constraints: ->(req) { Hostname.blog_site?(req.host) }
root 'landing#show'
```

**But I would like to have:**

``` ruby
root 'portfolio#show', constraints: ->(req) { Hostname.portfolio_site?(req.host) }
root 'blog#show',      constraints: ->(req) { Hostname.blog_site?(req.host) }
root 'landing#show'
```

Background: When an application has multiple `root` entries in same scope with different constraints, the current solution is to use `get '/'` instead of simply `root`. This is because when we try to add another entry of `root`, Rails complains there's already a route with `root` name.
The same doesn't happen with multiple `get`'s (or other matchers). So, I think it's fair that `root` also allows it since it's just a shortcut for a `get` internally.

PS: I couldn't find a spec related to allowing multiple `get`'s, `post`'s, etc with same path. If you know where is it, I could implement a similar spec for `root`.

### CODE REVIEW

**Comment by rafaelsales** [comment]
Anything needed from my end to have feedback here?

**Comment by seuros** [comment]
LGTM 
cc @rafaelfranca

**Comment by rafaelsales** [comment]
ping @rafaelfranca

**Comment by arthurnn** [comment]
Not sure about this honestly. the multiple `get '/',` look good to me already. I think this will add a few more bits of complexity to the code, that can already handle the use case described.
:-1: from me.

**Comment by rafaelsales** [comment]
But then routes API is inconsistent. You can have multiple `root` but with a "hack", using `get '/'`
Also, where this adds complexity to code? Instead, this makes the code consistent.

File: actionpack/lib/action_dispatch/routing/mapper.rb [hunk 1]
```diff
@@ -586,9 +587,8 @@ def with_default_scope(scope, &block)
           end
         end
 
-        # Query if the following named route was already defined.
```
Comment (rafaelfranca) [inline actionpack/lib/action_dispatch/routing/mapper.rb]: Why the documentation was removed?

**Comment by rafaelsales** [comment]
@rafaelfranca Removed that doc by mistake. Fixed and rebased

**Comment by rafaelsales** [comment]
ty
